3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -686,8 +648,98 @@ L'ambiente tridimensionale è realizzato mediante i componenti:
+
+- *Aggiornamenti e manutenzione*: la gestione di unico sistema semplifica la manutenzione e i numerosi aggiornamenti del codice durante lo sviluppo dettati dalla limitata esperienza del gruppo. Tuttavia, apportare modifiche consistenti può richiedere aggiornamenti in parti diverse del sistema.
+
+- *Deployment*: l'architettura monolitica semplifica il processo di deployment, dovendo gestire un unico sistema.
+
+== Deployment con Docker
+Il processo di deployment del software è gestito mediante l'utilizzo di Docker e Docker-compose. La scelta di utilizzare Docker è stata determinata dai seguenti fattori:
+
+- *Isolamento*: Docker permette di eseguire processi informatici in ambienti isolati chiamati container. Questo garantisce che il software funzioni in modo coerente e affidabile creando una base comune di sviluppo tra i Programmatori e garantendo che il prodotto software sia indipendente dall'ambiente di esecuzione;
+
+- *Portabilità*: Docker semplifica il processo di deployment del prodotto in diversi ambienti, garantendo che il software funzioni in modo coerente e affidabile su qualsiasi piattaforma;
+
+- *Semplicità di installazione e avvio*: impostata la configurazione dei Docker file e di Docker-compose, l'avvio e la gestione dei container è semplice e veloce.
+
+In conclusione, l'architettura monolitica si allinea perfettamente con le esigenze e le limitazioni del progetto, rendendola una scelta ragionata e valida. Questa scelta permette di concentrarsi sulle priorità chiave: sviluppare un prodotto funzionante e di alta qualità in tempi ragionevoli, pur mantenendo la flessibilità di apportare modifiche in base alle esigenze emergenti.
 
-== Database
+\
+=== Ambiente Docker
+I container Docker sono organizzati e gestiti mediante Docker-compose.
+
+Sono presenti due container:
+
+- *app*: container contenente l'applicazione web;
+  - *Immagine*: `node:20-alpine`;
+  - *Porta*: `3000`;
+  - *Dipendenza*: `postgres`;
+  - *Rete*: `webnet`.
+
+
+- *postgres*: container contenente il database PostgreSQL;
+  - *Immagine*: `postgres:16.2`;
+  - *Porta*: `5432`;
+  - *Dipendenza*: nessuna;
+  - *Rete*: `webnet`.
+
+I container sono all'interno della stessa rete `webnet` che utilizza il driver di rete `bridge`.
+
+#figure(
+  image("./imgs/docker-env.png", width: 70%),
+  caption: [
+    Ambiente Docker
+  ],
+)
+
+\
+== Deployment e installazione
+
+=== Download di WMS3 <download_WMS3>
+
+#figure(
+  table(
+    columns: (auto, 1fr),
+    [*Modalità*], [*Source*],
+    [Link diretto (consigliato)], [#link("https://github.com/Error-418-SWE/WMS3/releases")],
+    [Shell], [`git clone git@github.com:Error-418-SWE/WMS3.git`],
+    [Shell], [`git clone https://github.com/Error-418-SWE/WMS3.git`]
+  ),
+  caption: "Modalità di download WMS3."
```

Comment:
Rimuovere il punto a fine caption. È presente anche in altre caption di tabelle o figure, quindi dai un occhio veloce

Reply:
Cosa significa "link diretto consigliato"?

Reply:
Consigliato perchè è il più immediato ma... si beh non cambia nulla ahah